src/libraries/Microsoft.PowerFx.Core/Public/Values/TableValue.cs

```diff
@@ -125,25 +126,32 @@ protected virtual async Task<DValue<RecordValue>> PatchCoreAsync(RecordValue bas
         /// </summary>
         /// <param name="baseRecord">A record to modify.</param>
         /// <param name="changeRecord">A record that contains properties to modify the base record.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>The updated record.</returns>
-        public async Task<DValue<RecordValue>> PatchAsync(RecordValue baseRecord, RecordValue changeRecord)
+        public async Task<DValue<RecordValue>> PatchAsync(RecordValue baseRecord, RecordValue changeRecord, CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             var recordType = Type.ToRecord();
 
             // IR has already resolved to logical names because of 
             // RequiresDataSourceScope, ArgMatchesDatasourceType on function.
-            return await PatchCoreAsync(baseRecord, changeRecord);
+            return await PatchCoreAsync(baseRecord, changeRecord, cancellationToken);
         }
 
         public override object ToObject()
         {
             if (IsColumn)
             {
-                var array = Rows.Select(val =>
+                var array = Rows.Select(async val =>
                 {
                     if (val.IsValue)
                     {
-                        return val.Value.Fields.First().Value.ToObject();
+                        await foreach (var field in val.Value.GetFieldsAsync(CancellationToken.None))
+                        {
+                            return field.Value.ToObject();
+                        }
+
+                        throw new NotImplementedException();
```

Comment:
>throw new NotImplementedException(); [](http://example.com/codeflow?start=24&length=36)

Before it would not throw anything, now it does... Do we have a test that hits this codepath?

Reply:
It would have thrown if First() would have failed 
That said, it's safer to remove it, I agree